### PR TITLE
test/cli: content blocking tests for CARs

### PR DESCRIPTION
**Pull request by hacdias**

### CODE REVIEW

File: test/cli/content_blocking_test.go [hunk 1]
```diff
@@ -168,18 +193,18 @@ func TestContentBlocking(t *testing.T) {
...
-
+		for _, format := range []string{"", "raw", "car"} { // CARs should also fail immediately since we access the file directly.
+			// Confirm that denylist is active for every content path in 'testCases' with different formats
+			gwTestName := fmt.Sprintf("Gateway denies %s with format %s", testCase.name, format)
+			t.Run(gwTestName, func(t *testing.T) {
+				resp := client.Get(testCase.path + "?format=" + format)
+				assert.Equal(t, http.StatusGone, resp.StatusCode, statusExpl)
+				assert.Contains(t, resp.Body, blockedMsg, bodyExpl)
+			})
+		}
```
Comment (hacdias) [inline test/cli/content_blocking_test.go:204]: The `CAR` times out on both .26 and current PR. Not sure exactly why.

File: test/cli/content_blocking_test.go [hunk 2]
```diff
@@ -98,6 +101,28 @@ func TestContentBlocking(t *testing.T) {
...
+		resp := client.Get("/ipfs/" + parentDirCID + "?format=car")
+		require.Equal(t, http.StatusOK, resp.StatusCode)
+
+		bs, err := cbs.NewReadOnly(strings.NewReader(resp.Body), nil)
+		require.NoError(t, err)
+
+		has, err := bs.Has(context.Background(), cid.MustParse(blockedCIDInParentDirCID))
+		require.NoError(t, err)
+		require.False(t, has)
+	})
```
Comment (hacdias) [inline test/cli/content_blocking_test.go:124]: Fail on both 0.26 and `master`.

File: test/cli/content_blocking_test.go [hunk 3]
```diff
@@ -299,5 +324,17 @@ func TestContentBlocking(t *testing.T) {
...
+			t.Parallel()
+			resp, err := libp2pClient.Get(fmt.Sprintf("/ipfs/%s?format=car", blockedCID))
+			require.NoError(t, err)
+			defer resp.Body.Close()
+			assert.Equal(t, http.StatusGone, resp.StatusCode, statusExpl)
+			body, err := io.ReadAll(resp.Body)
+			require.NoError(t, err)
+			assert.NotEqual(t, string(body), "directly blocked file content")
+			assert.Contains(t, string(body), blockedMsg, bodyExpl)
+		})
```
Comment (hacdias) [inline test/cli/content_blocking_test.go:338]: Passes on both 0.26 and `master`.

**Review by lidel — COMMENTED** [review]
I also was not able to find a case which looks like a regression since 0.26.

Bit late on my end, but iiuc the bug described in https://github.com/ipfs/kubo/issues/10361 was always present.


I propose we:
1. proceed with https://github.com/ipfs/kubo/issues/10306 (RC2)
2. close this PR draft
3. merge cleanup from https://github.com/ipfs/kubo/pull/10360 instead
4. fix https://github.com/ipfs/kubo/issues/10361 in next iteration (Kubo 0.28)

File: test/cli/content_blocking_test.go (hunk 2 above)
Comment (lidel) [reply test/cli/content_blocking_test.go]: Filled https://github.com/ipfs/kubo/issues/10361

File: test/cli/content_blocking_test.go (hunk 1 above)
Comment (lidel) [reply test/cli/content_blocking_test.go:211]: It feels like some tests fail due to the same bug as https://github.com/ipfs/kubo/issues/10361

**Pull request closed without merging**